app/models/poll.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+class Poll < ActiveRecord::Base
+  has_many :options, dependent: :destroy
+
+  validates :title, presence: true
+
+  after_touch :ensure_status_is_correct
+
+  def make_ready
```

Comment:
Because ready != published, ithink it will be draft->ready->published->(may be in progress)->finished | closed (finished succesfull or closed by force)->deleted